SemVer.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Created by Anita on 07/08/2016.
+ */
+
+myModuleI = {
+    SemVer: function (major, minor, revision) {
+        if (major < 0 || minor < 0 || revision < 0) {
+            throw new Error('Error: negative number' );
+        }else {
+            this.major = major;
+            this.minor = minor;
+            this.revision = revision;
+        }
+        /* Code to see if major is valid, how implements that? And then, testing:
+
+        var regex = /^[1-9][0-9][0-9]$|^[1-9][0-9]$|^[1-9]$/;
+        if(regex.test(major)){
+            return major;
+        }else{
+            return 'Value for major incorrect';
+        }
+
+        */
+
+        this.convertToString = function () {
+            var major = this.major.toString();
+            var minor = this.minor.toString();
+            var revision = this.revision.toString();
+
+            var result = major + '.' + minor + '.' + revision;
+
+            return result;
+        };
+
+    },
+
+    parse: function (string) {
+
+        var reg = /^(\d{1,3})\.(\d{1,3})\.(\d{1,3})$/.test(string);
+
+        if (!reg) {
+            throw new Error('Error: Invalid String');
+        } else {
+            var majorInt = Number(/^(\d{1,3})/.exec(string)[1]);
```

Comment:
Why create 3 new regex, when we can obtain via subgroups the values on the original regex ?
Also try not to do many actions in one line (ex: regex match + new Number)